cpp/arcticdb/storage/s3/nfs_backed_storage.cpp

```diff
@@ -197,18 +198,24 @@ NfsBackedStorage::NfsBackedStorage(const LibraryPath &library_path, OpenMode mod
     } else {
         s3_client_ = std::make_unique<s3::RealS3Client>(s3::get_aws_credentials(conf), s3::get_s3_config(conf), Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy::Never, false);
     }
-    if (!conf.prefix().empty()) {
-        ARCTICDB_DEBUG(log::version(), "prefix found, using: {}", conf.prefix());
+
+    if (conf.prefix().empty()) {
+        ARCTICDB_DEBUG(log::version(), "prefix not found, will use {}", root_folder_);
+    } else if (conf.use_raw_prefix()) {
+        ARCTICDB_DEBUG(log::version(), "raw prefix found, using: {}", conf.prefix());
+        root_folder_ = conf.prefix();
+    } else {
         auto prefix_path = LibraryPath::from_delim_path(conf.prefix(), '.');
         root_folder_ = object_store_utils::get_root_folder(prefix_path);
-    } else {
-        ARCTICDB_DEBUG(log::version(), "prefix not found, will use {}", root_folder_);
+        ARCTICDB_DEBUG(log::version(), "parsed prefix found, using: {}", root_folder_);
     }
+
     // When linking against libraries built with pre-GCC5 compilers, the num_put facet is not initalized on the classic locale
     // Rather than change the locale globally, which might cause unexpected behaviour in legacy code, just add the required
     // facet here
     std::locale locale{ std::locale::classic(), new std::num_put<char>()};
     (void)std::locale::global(locale);
+    s3_api_.reset();
```

Comment:
This was missing before and caused this issue when running the tests with a debug build, https://github.com/aws/aws-sdk-cpp/issues/1451